File: IrrgartenP3/IrrgartenJava/src/Tests/TestP3.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Tests;
/**
 *
 * @author moham
 */
import irrgarten.*;

import irrgarten.UI.TextUI;
import irrgarten.controller.Controller;

public class TestP3 {
    public static void main(String[] args) {
        // Configuración del juego: número de jugadores, filas y columnas del laberinto
        int nPlayers = 2;  // Número de jugadores (puedes cambiarlo)
        int nRows = 5;     // Número de filas del laberinto
        int nCols = 5;     // Número de columnas del laberinto

        // Crear la instancia del juego
        Game game = new Game(nPlayers, nRows, nCols);

        // Crear la vista
        TextUI textUI = new TextUI();

        // Crear el controlador
        Controller controller = new Controller(game, textUI);

        // Iniciar el juego
        if(!game.finished()) controller.play();
    }
}

File: IrrgartenP4/irrgartenRuby/irrgarten/test_p3.rb
# frozen_string_literal: true
require_relative 'game'
require_relative 'monster'
require_relative 'player'
require_relative 'labyrinth'
require_relative 'game'
require_relative 'weapon'
require_relative 'shield'
require_relative 'dice'
require_relative 'game_state'
require_relative '../UI/textUI'
require_relative '../Control/controller'

module Main
  class MainTest
    def self.main
      vista=UI::TextUI.new
      juego = Irrgarten::Game.new(2,5,5)
      controlador=Control::Controller.new(juego,vista)
      controlador.play
    end
  end
end

Main::MainTest.main